### PR TITLE
[Line Heights] Add textHeightBehavior to SelectableText.

### PR DESCRIPTION
**Pull request by johnsonmh**
## Description

This PR exposes the `textHeightBehavior` parameter on `SelectableText` widgets, so that users of `SelectableText` can specify the text height behavior.

## Tests

I added the following tests:

* Default text height behavior is still the same.
* Text height behavior can be set on `SelectableText` widgets. 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

File: packages/flutter/lib/src/material/selectable_text.dart [hunk 1]
```diff
@@ -406,6 +408,9 @@ class SelectableText extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.scrollPhysics}
   final ScrollPhysics scrollPhysics;
 
+  /// {@macro flutter.painting.textPainter.textHeightBehavior}
```
Comment (JoseAlba) [inline packages/flutter/lib/src/material/selectable_text.dart:411]: Did you define this one and forgot to copy it over or does it already exist ?

**Review by JoseAlba — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

File: packages/flutter/lib/src/material/selectable_text.dart (hunk 1 above)
Comment (GaryQian) [reply packages/flutter/lib/src/material/selectable_text.dart:411]: Did you mean `{@macro flutter.dart:ui.textHeightBehavior}`, which is what we are using as the macro in `text_painter.dart`

**Review by GaryQian — APPROVED** [review]
Comments nit, otherwise LGTM

**Review by rami-a — APPROVED** [review]

File: packages/flutter/lib/src/material/selectable_text.dart (hunk 1 above)
Comment (johnsonmh) [reply packages/flutter/lib/src/material/selectable_text.dart]: Thanks for the spot, I updated all the macro references to `{@macro flutter.dart:ui.textHeightBehavior}`

**Pull request merged**